src/acl/Checklist.cc

```diff
@@ -137,8 +136,8 @@ ACLChecklist::goAsync(AsyncState *state)
     ++asyncLoopDepth_;
 
     asyncStage_ = asyncStarting;
-    changeState(state);
-    state->checkForAsync(this); // this is supposed to go async
+    changeState(&starter);
+    starter(*this, acl); // this is supposed to go async
```

Comment:
Are there any cases where `starter` cannot be a virtual ACL::startAsync() method? In other words, are there any cases where the same ACL class type is used with two _different_ AsyncStarter pointers, making a single virtual method an inappropriate entry point?

Reply:
> I see one such case in external_acl.cc with two different functions, ACLExternal::StartLookup and ACLIdent::StartLookup.

But ACLExternal and ACLIdent are _different_ ACL classes. Could not each of them override a virtual ACL::startAsync() method, providing a class-specific definition?

Reply:
> But ACLExternal and ACLIdent are different ACL classes.

Yes, but we pass an ACLExternal object (as this) to both goAsync() calls:
```
ch->goAsync(StartLookup, *this)
ch->goAsync(ACLIdent::StartLookup, *this);
```

Reply:
Yes, if external_acl_type is configured with %ui, makeExternalAclKey() returns nullptr.
Squid aborts in this case even without extra assertions:
```
2023/10/27 02:39:28| ALE missing IDENT
    current master transaction: master53
2023/10/27 02:39:48| assertion failed: FilledChecklist.cc:273: "!rfc931[0]"
    current master transaction: master53
```
This happens with PR Squid and some (not fresh) master Squid.

Reply:
Great, let's do a prerequisite internal PR that removes that (broken) "IDENT from external ACL" functionality and simplifies external ACL code.

Reply:
Update: Upon closer inspection, those assertions are not related to this change request. We will fix them in separate PRs.